fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/indexes/PermutedMinMaxIndexTest.java

```diff
@@ -201,6 +210,443 @@ public void tie() {
+            recordStore.deleteRecord(Tuple.from(11));
+
+            if (min) {
+                assertEquals(Arrays.asList(
+                        Tuple.from(5, 1),
+                        Tuple.from(5, 3),
+                        Tuple.from(5, 5),
+                        Tuple.from(7, 2),
+                        Tuple.from(7, 4)
+                ), scanGroup(Tuple.from("yes"), false));
+
+                assertEquals(Arrays.asList(
+                        Tuple.from(3, 5),
+                        Tuple.from(5, 2),
+                        Tuple.from(5, 4),
+                        Tuple.from(7, 1),
+                        Tuple.from(7, 3)
+                ), scanGroup(Tuple.from("no"), false));
+
+            } else {
+                assertEquals(Arrays.asList(
+                        Tuple.from(5, 1),
+                        Tuple.from(5, 3),
+                        Tuple.from(7, 2),
+                        Tuple.from(7, 4),
+                        Tuple.from(9, 5)
+                ), scanGroup(Tuple.from("yes"), false));
+
+                assertEquals(Arrays.asList(
+                        Tuple.from(5, 2),
+                        Tuple.from(5, 4),
+                        Tuple.from(7, 1),
+                        Tuple.from(7, 3),
+                        Tuple.from(7, 5)
+                ), scanGroup(Tuple.from("no"), false));
+            }
+
+            commit(context);
+        }
+    }
+
+    @ParameterizedTest(name = "repeatedValue[min={0}]")
+    @BooleanSource
+    void repeatedValue(boolean min) {
+        // Index on:
+        //    min/max(repeater) GROUP BY str_value_indexed, num_value_2 ORDER BY str_value_indexed, min/max(repeater), num_value_2
+        // Note that all values of repeater will always come from a single group, so at most one entry in the permuted space should ever be updated on record insert
+        final RecordMetaDataHook hook = hook(min,
+                Key.Expressions.field("repeater", KeyExpression.FanType.FanOut).groupBy(Key.Expressions.field("str_value_indexed"), Key.Expressions.field("num_value_2")),
+                1);
+
+
+        try (FDBRecordContext context = openContext()) {
+            openSimpleRecordStore(context, hook);
+
+            saveRecord(1, "yes", 1, 0, 1, 2, 3, 4, 5);
+            saveRecord(2, "no", 1, 0, 1, 3, 5);
+
+            assertEquals(Collections.singletonList(
+                    Tuple.from(min ? 1 : 5, 1)
+            ), scanGroup(Tuple.from("yes"), false));
+
+            assertEquals(Collections.singletonList(
+                    Tuple.from(min ? 1 : 5, 1)
+            ), scanGroup(Tuple.from("no"), false));
+
+            assertEquals(Arrays.asList(
+                    Tuple.from(1, 1),
+                    Tuple.from(2, 1),
+                    Tuple.from(3, 1),
+                    Tuple.from(4, 1),
+                    Tuple.from(5, 1)
+            ), scanValue(Tuple.from("yes", 1), false));
```

Comment:
There's a version of this that I almost wrote where the ordinary tree index only gets one entry per group per record. Essentially, it would take what is in their name, but instead of calling `super.updateIndexKeys` with _all_ of the entries, it would instead call `updateOneKeyAsync` within a loop only with the most extreme entry per group. I added this check to assert that we're not doing that, because this would violate the way the planner currently treats the ordinary tree index (i.e., it expects it to behave just like a value index on the whole key)